src/main/java/ru/yandex/practicum/filmorate/storage/dao/DBFilmStorage.java

```diff
@@ -0,0 +1,183 @@
+
+        if (!film.getGenres().isEmpty()) {
+            genreService.addFilmGenres(film.getId(), film.getGenres());
+        }
+        if (film.getLikes() != null) {
+            for (Integer userId : film.getLikes()) {
+                addLike(film.getId(), userId);
+            }
+        }
+        return getFilm(id);
+    }
+
+    @Override
+    public Film updateFilm(Film film) {
+    String sqlQuery = "update FILM " +
+            "set NAME = ?, DESCRIPTION = ?, RELEASEDATE = ?, DURATION = ?, RATE = ? ,RATINGID = ? " +
+            "where FILMID = ?";
+        jdbcTemplate.update(sqlQuery,
+                film.getName(),
+                film.getDescription(),
+                film.getReleaseDate(),
+                film.getDuration(),
+                film.getRate(),
+                film.getMpa().getId(),
+                film.getId());
+
+        genreService.deleteFilmGenres(film.getId());
+        if (!film.getGenres().isEmpty()) {
+            genreService.addFilmGenres(film.getId(), film.getGenres());
+        }
+
+        if(film.getLikes() != null) {
+            for (Integer userId : film.getLikes()) {
+                addLike(film.getId(), userId);
+            }
+        }
+        return getFilm(film.getId());
+    }
+
+    @Override
+    public boolean deleteFilm(Film film) {
+        String sqlQuery = "delete from FILM where FILMID = ?";
+        jdbcTemplate.update(sqlQuery, film.getId());
+        return true;
+    }
+
+    @Override
+    public boolean addLike(int filmId, int userId) {
+        String sql = "select * from LIKES where USERID = ? and FILMID = ?";
+        SqlRowSet existLike = jdbcTemplate.queryForRowSet(sql, userId, filmId);
+        if (!existLike.next()) {
+            String setLike = "insert into LIKES (USERID, FILMID) values (?, ?) ";
+            jdbcTemplate.update(setLike, userId, filmId);
+        }
+        SqlRowSet rs = jdbcTemplate.queryForRowSet(sql, userId, filmId);
+        log.info(String.valueOf(rs.next()));
+        return rs.next();
+    }
+
+    @Override
+    public boolean deleteLike(int filmId, int userId) {
+        String deleteLike = "delete from LIKES where FILMID = ? and USERID = ?";
+        jdbcTemplate.update(deleteLike, filmId, userId);
+        return true;
+    }
+
+    @Override
+    public Collection<Film> getMostPopularFilms(int count) {
+        String sqlMostPopular = "select count(L.LIKEID) as likeRate" +
+                ",FILM.FILMID" +
+                ",FILM.NAME ,FILM.DESCRIPTION ,RELEASEDATE ,DURATION ,RATE ,R.RATINGID, R.NAME, R.DESCRIPTION from FILM " +
+                "left join LIKES L on L.FILMID = FILM.FILMID " +
+                "inner join RATINGMPA R on R.RATINGID = FILM.RATINGID " +
+                "group by FILM.FILMID " +
+                "ORDER BY likeRate desc " +
+                "limit ?";
+        Collection<Film> films = jdbcTemplate.query(sqlMostPopular, (rs, rowNum) -> makeFilm(rs), count);
+        return films;
```

Comment:
Переменная `films` лишняя, так как нигде больше не используется, лучше сразу возвращать результат `query`:
```suggestion
        return jdbcTemplate.query(sqlMostPopular, (rs, rowNum) -> makeFilm(rs), count);
```